internals/overlord/state/identities.go

```diff
@@ -83,10 +83,6 @@ func (d *Identity) validateAccess() error {
 
 	switch d.Access {
 	case AdminAccess, ReadAccess, MetricsAccess, UntrustedAccess:
-		// Check basic type access level.
-		if d.Basic != nil && d.Access != MetricsAccess {
-			return fmt.Errorf("basic identity can only have %q access, got %q", MetricsAccess, d.Access)
-		}
```

Comment:
@benhoyt I have a question here:

This check, and its location feels a bit problematic. We would like to use basic with with HTTPS. Currently the source of the http.Request is not known here, but can maybe be passed in during ```IdentityFromInputs``` from the daemon.

The other option (which I think may be easier to implement, is to move this level of checking into the access checker on the daemon side, where we have all the connection details.

I have not spent a lot of time thinking about the options, but what I would like to say is:

1. Basic Auth is always OK over HTTPS (even for Admin)
2. Basic Auth is OK over HTTP, but only for MetricsAccess.

Any hints on which direction you might go ?

Reply:
Yeah, we put this in place (in discussion with Gustavo) till we had HTTPS support, so that the username/password going over unencrypted HTTP was only good for metrics. Once we have HTTPS, I think your proposal sounds good: that basic auth is always OK over HTTPS, but only for metrics over HTTP.

In terms of implementation, without digging in I don't have a strong opinion. I'd be happy to review the simpler (in the daemon) implementation you propose and go from there.

Reply:
Implemented. Resolving this comment.